### DISCUSSION TITLE
Ability to add commands for chatbots and autocomplete for them

### DISCUSSION THREAD

**Discussion by x9136** [Chatterino/chatterino2#5371] [Ideas] [answered]
Let everyone make a list of commands for chatbots and show them for autocomplete. As for autocomplete, #4935 looks useful, but the same thing that works for Emotes will do.

This feature will allow you to call commands faster, you will not have to remember how to write the right command, and there will be fewer errors.

![manual_commands](https://github.com/Chatterino/chatterino2/assets/60105060/fcf6e45d-8ea2-44f7-95a7-4a9112f02e3f)

**Comment by Nerixyz** [comment]
https://github.com/Chatterino/chatterino2/pull/3209 tries to implement autocompletion for bots. I suppose plugins could provide this kind of functionality in the future (or for some bots today).

**Reply by x9136** [reply]
I have suggested a simpler solution in case someone wants to solve the issue much earlier.

**Reply by Mm2PL** [reply, marked as answer]
A plugin calling an HTTP API is probably the more versatile solution to this. I would highly prefer that solution over baking in per-channel commands simply because plugins can handle more weird edge cases like mod-only commands, arguments to commands, sub commands.
